bin/install-habitat

```diff
@@ -0,0 +1,101 @@
+#
+# Copyright:: Copyright 2017 Chef Software, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+#
+# This script is based off of https://gist.github.com/cosimo/3760587
+
+set -eou pipefail
+
+function usage() {
+  cat <<DOC
+Usage: ${0##*/} [-u USER] [-v HAB_VERSION] [-c HAB_CHANNEL] -h
+
+Install VERSION of habitat from CHANNEL as the USER user.
+
+OPTIONS:
+  -h            Show this message.
+  -u USER       The user you want to run hab commands as. (default: user in '/var/opt/ci-studio-common/.hab-user' file or 'root')
+  -v VERSION    Which version of Habitat you wish to install. (default: version in '.hab-version' file)
+  -c CHANNEL    The channel from which you wish to install Habitat. (default: stable)
+DOC
+}
+
+VERSION=$(curl -sL https://raw.githubusercontent.com/chef/ci-studio-common/master/.hab-version | xargs echo -n)
+CHANNEL=stable
+
+HAB_USER_FILE="/var/opt/ci-studio-common/.hab-user"
+if [[ -f $HAB_USER_FILE ]]; then
+  USER=$(cat $HAB_USER_FILE)
+else
+  USER=$(whoami)
+fi
+
+while getopts ":hcvu:" opt; do
+  case "${opt}" in
+    c)
+      CHANNEL="${OPTARG}"
+      ;;
+    v)
+      VERSION="${OPTARG}"
+      ;;
+    u)
+      USER="${OPTARG}"
+      ;;
+    h)
+      usage
+      exit 0
+      ;;
+    \?)
+      usage
+      exit 1
+      ;;
+    *)
+      ;;
+  esac
+done
+
+echo "--- Installing Habitat"
+
+echo "Setting USER as '$USER'"
+mkdir -p "/var/opt/ci-studio-common"
+echo "$USER" > "$HAB_USER_FILE"
+
+echo "Creating /hab directory"
+mkdir -p /hab
+
+echo "Granting $USER access to /hab"
+chown -R "root:$USER" /hab
+chmod g+rwx /hab
+
+echo "Running habitat curl|bash"
+curl -sL https://raw.githubusercontent.com/habitat-sh/habitat/master/components/hab/install.sh | bash -s -- -v "$VERSION" -c "$CHANNEL"
+
+hab_bin=$(command -v hab)
+echo "Granging $USER access to $hab_bin"
+chown -R "root:$USER" "$hab_bin"
+
+if [[ "$USER" != "root" ]]; then
+  echo "Granting '$USER' sudo permissions to run $hab_bin and $install_hab_bin"
+  install_hab_bin=$(command -v install-habitat)
+  file-mod append-if-missing "$USER ALL=NOPASSWD:SETENV: $hab_bin" "/etc/sudoers.d/$USER"
+  file-mod append-if-missing "$USER ALL=NOPASSWD:SETENV: $install_hab_bin" "/etc/sudoers.d/$USER"
+  chmod 440 "/etc/sudoers.d/$USER"
+fi
+
+# TODO: There is probably some work here that can be done to programatically determine whether we need to update the
+# launcher service, but that is a task for another time.
```

Comment:
I think that will be hard to determine programmatically as the launcher service is indented to be updated as rarely, and even if there are changes, that doesn't necessarily mean it should be. For example, we may improve the help output or other nonfunctional things, but we only recommend updating launcher when we fix bugs since it requires stopping all services and taking downtime.